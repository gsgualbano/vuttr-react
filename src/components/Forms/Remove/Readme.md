#### Formulário para remoção de ferramentas.

Exemplo:

```js
const tool = {
  id: 1,
  title: "Google",
  description: "Search Website",
  link: "https://google.com",
  tags: ["search", "site", "google"]
};

const setShowModal = () => {};

<Remove tool={tool} setShowModal={setShowModal} />;
```
