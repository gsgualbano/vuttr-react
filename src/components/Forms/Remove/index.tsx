import React from 'react';

import { bindActionCreators, Dispatch } from 'redux';
import { connect } from 'react-redux';

import * as ToolsActions from '~/store/ducks/tools/actions';
import { Tool } from '~/store/ducks/tools/types';

import {
 Container, Header, Content, Bottom, Text, IconContainer, CloseIcon,
} from './styles';
import PrimaryButton from '~/components/Common/PrimaryButton';
import DangerButton from '~/components/Common/DangerButton';

interface DispatchProps {
  /** saga action remove tool */
  removeTool(id: number): void;
}

/**
 * Props do RemoveForm
 */
interface OwnProps {
  /** objeto com informações sobre a ferramenta como id, title, description, link, tags */
  tool: Tool;
  /** dispatcher para mostrar o modal */
  setShowModal: React.Dispatch<React.SetStateAction<boolean>>;
}

type Props = OwnProps & DispatchProps;

/**
 * Remove Form.
 */
const RemoveForm = ({ tool, removeTool, setShowModal }: Props): JSX.Element => {
  const handleCancelButton = (): void => {
    setShowModal(false);
  };

  const handleRemoveButton = (id: number): void => {
    removeTool(tool.id);
    setShowModal(false);
  };

  return (
    <Container>
      <Header>
        <IconContainer>
          <CloseIcon />
        </IconContainer>
        Remove tool
      </Header>
      <Content>
        <Text>{`Are you sure you want to remove ${tool.title} ?`}</Text>
      </Content>
      <Bottom>
        <PrimaryButton onClick={handleCancelButton}>Cancel</PrimaryButton>
        <DangerButton onClick={() => handleRemoveButton(tool.id)}>Yes, remove</DangerButton>
      </Bottom>
    </Container>
  );
};

const mapDispatchToProps = (dispatch: Dispatch) => bindActionCreators(ToolsActions, dispatch);

export default connect(
  null,
  mapDispatchToProps,
)(RemoveForm);
