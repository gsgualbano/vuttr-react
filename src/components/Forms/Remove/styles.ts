import styled from 'styled-components';
import { ReactComponent as Icon } from '~/assets/icons/Icon-Close-2px.svg';

export const Container = styled.div``;

export const Header = styled.div`
  display: flex;
  flex-direction: row;
  font-weight: 600;
  margin: 10px 0 20px;
`;

export const IconContainer = styled.div`
  height: 100%;
  justify-content: center;
  align-items: center;
`;

export const CloseIcon = styled(Icon)`
  width: 14px;
  height: 14px;
  margin-right: 10px;

  g > path.a {
    stroke: #f95e5a;
    stroke-width: 10px;
  }
`;

export const Content = styled.div`
  margin: 20px 0 30px;
`;

export const Text = styled.p`
  color: #8f8a9b;
`;

export const Bottom = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
`;
