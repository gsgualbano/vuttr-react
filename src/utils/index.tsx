/* eslint-disable max-len */
import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

import { render } from '@testing-library/react';

import { ThemeProvider } from 'styled-components';
import { reducer } from '../store/ducks/tools';

interface Theme {
  colors: {
    background: string;
    textColor: string;
    cardBackground: string;
    cardBorderColor: string;
    modalBackground: string;
    modalBorderColor: string;
    linkColor: string;
    linkVisitedColor: string;
    buttonOutlineBackground: string;
    InputBackground: string;
    InputBorderColor: string;
  };
}

/** redux mock to render components that depends on the redux */
export const renderWithRedux = (
  component: JSX.Element,
  { initialState, store = createStore(reducer, initialState) }: any = {},
) => ({
  ...render(<Provider store={store}>{component}</Provider>),
  store,
});

/** render components thats depends on the theme provider */
export const renderWithTheme = (component: JSX.Element, theme: Theme) => render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

/** render components thats depends on the redux and theme provider */
export const renderWithReduxAndTheme = (
  component: JSX.Element,
  theme: Theme,
  { initialState, store = createStore(reducer, initialState) }: any = {},
) => ({
  ...render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>{component}</Provider>
    </ThemeProvider>,
  ),
  store,
});
