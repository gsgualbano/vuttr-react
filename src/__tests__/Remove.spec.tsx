import React from 'react';
import { renderWithRedux } from '~/utils';
import Remove from '~/components/Forms/Remove';
import 'jest-styled-components';

describe('<Remove />', () => {
  const tool = {
    id: 1,
    title: 'Google',
    link: 'http://google.com',
    description: 'Search Website',
    tags: ['search', 'engine', 'website'],
  };

  it('render correctly', () => {
    const { asFragment } = renderWithRedux(<Remove setShowModal={() => {}} tool={tool} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
