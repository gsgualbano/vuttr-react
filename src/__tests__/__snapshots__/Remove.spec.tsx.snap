// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Remove /> render correctly 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-weight: 600;
  margin: 10px 0 20px;
}

.c1 {
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  width: 14px;
  height: 14px;
  margin-right: 10px;
}

.c2 g > path.a {
  stroke: #f95e5a;
  stroke-width: 10px;
}

.c3 {
  margin: 20px 0 30px;
}

.c4 {
  color: #8f8a9b;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-size: 16px;
  color: #ffffff;
  background: #365df0;
  padding: 5px 10px;
  border-radius: 5px;
  border: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6:hover {
  background: #2f55cc;
}

.c6:click {
  background: #244aa8;
  background-color: #244aa8;
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #ffffff;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  padding: 5px 10px;
  border: none;
  margin-left: 5px;
  background: #f95e5a;
  border-radius: 5px;
}

.c7:hover {
  background: #cc4c4c;
}

.c7:click {
  background: #a53f3f;
}

@media (max-width:768px) {
  .c6 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c7 {
    margin-left: 5px;
    width: 100%;
  }
}

<div
    class=""
  >
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <svg
          class="c2"
        >
          Icon-Close-2px.svg
        </svg>
      </div>
      Remove tool
    </div>
    <div
      class="c3"
    >
      <p
        class="c4"
      >
        Are you sure you want to remove Google ?
      </p>
    </div>
    <div
      class="c5"
    >
      <button
        class="c6"
        data-testid="primary-button"
        type="button"
      >
        Cancel
      </button>
      <button
        class="c7"
        data-testid="danger-button"
        type="button"
      >
        Yes, remove
      </button>
    </div>
  </div>
</DocumentFragment>
`;
